frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -88,7 +92,29 @@ fn complete_modules(decl: impl Iterator<Item = ModuleDeclaration>) -> syn::Resul
 				return Err(err);
 			}
 
+			// Get if pallet use struct Module or Pallet.
+			let module_part = module.module_parts.iter().find(|part| part.name() == "Module");
+			let pallet_part = module.module_parts.iter().find(|part| part.name() == "Pallet");
+			let pallet_or_module = match (module_part, pallet_part) {
```

Comment:
Before this was no requirement, why is it now required to have this?

Actually it was a very bad idea back in the days to introduce `Module`, as we enforce this anyway. Aka, a pallet/module without the struct doesn't work.

Reply:
ah I see, I used this part `Module` to identify if we should use `Module` or `Pallet`. But actually it is not its usage.
if part `Module` exist then just metadata will contains some module associated metadata.

I think it is indeed better not to change this. Instead I should have new macro generating a type alias to `Module`. and Have construct_runtime use `Module` as before.
Later when we deprecate old macros we can factorize it but a type alias here is not an issue.

Reply:
I make pallet generating a type alias `Module` and revert all changes to construct_runtime.

I think it is cleaner too, refactoring of construct_runtime can be done later and differently